controllers/token_scorer/token_scorer.py

```diff
@@ -23,19 +24,67 @@
 
 
 class Point(NamedTuple):
-    # We are only tracking tokens on a 2D plane, with Y as north
+    # We are only tracking tokens on a 2D plane, with Y as north.
+    # Values are in meters.
     x: float
     y: float
 
 
-SHIP_SCORING_ZONE = (Point(-0.75, 0.5), Point(0.75, -3))
-ZONE_0_STACK_SCORING_ZONE = (Point(-0.75, 0.5), Point(0.75, -0.5))
-ZONE_1_STACK_SCORING_ZONE = (Point(-0.75, -2), Point(0.75, -3))
+class ScoringLocation(enum.Enum):
+    """
+    A rectangular axis-aligned location used for scoring.
+
+    The enum is ordered from least specific location to most specific.
+    """
+
+    def __init__(self, points: int, owner: Owner, coordinates: Tuple[Point, Point]) -> None:
+        self._points = points
+        self.owner = owner
+        self._coordinates = coordinates
+
+        self.x_min = min(x for x, _ in self._coordinates)
+        self.x_max = max(x for x, _ in self._coordinates)
+        self.y_min = min(y for _, y in self._coordinates)
+        self.y_max = max(y for _, y in self._coordinates)
```

Comment:
I feel like it might be nice to factor this out to an AABB (axis-aligned bounding box) class but don't feel strongly.

Reply:
Perhaps a verification step to make sure that the coördinates _do_ in fact specify a rectangular bounding box?

Reply:
> Perhaps a verification step to make sure that the coördinates _do_ in fact specify a rectangular bounding box?

How can they not do? We're only accepting four input coordinates. Or are you suggesting that we might get two points on an axis-aligned line?

I did nearly copy the existing `Rectangle` class from the vision stuff, but didn't feel it was quite worth it.